src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -94,6 +94,16 @@ internal abstract partial class AnalyzerDriver : IDisposable
         /// </summary>
         private ImmutableHashSet<DiagnosticAnalyzer> _unsuppressedAnalyzers;
 
+        /// <summary>
+        /// Set of unsuppressed analyzers that report non-configurable diagnostics that cannot be suppressed with end user configuration. 
+        /// </summary>
+        private ImmutableHashSet<DiagnosticAnalyzer> _nonConfigurableAnalyzers;
```

Comment:
This file is a mess. How can we clean it up?

Reply:
Feel free to file a tracking issue for any cleanup work that you would like to propose.

Reply:
https://github.com/dotnet/roslyn/issues/38958